File: src/main/java/org/dedda/bratwurst/lang/io/file/FileExists.java
package org.dedda.bratwurst.lang.io.file;

import org.dedda.bratwurst.lang.*;
import org.dedda.bratwurst.lang.scope.Scope;

import java.io.File;

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
public class FileExists extends BWExpression {

    public final String variableName;
    private boolean exists;

    public FileExists(int lineNumber, String variableName) {
        super(lineNumber);
        this.variableName = variableName;
    }

    @Override
    public BWObject getValue() {
        return new BWInteger(getIntValue());
    }

    @Override
    public int getIntValue() {
        return exists ? 1 : 0;
    }

    @Override
    public String getValueType() {
        return "integer";
    }

    @Override
    public void run(Scope scope) {
        BWVariable variable = scope.getVariable(variableName);
        if (!(variable.getValue() instanceof BWString)) {
            throw new RuntimeException("variable not of type string!");
        }
        String fileName = ((BWString) variable.getValue()).getStringValue();
        File file = new File(fileName);
        this.exists = file.exists();
    }
}

File: src/main/java/org/dedda/bratwurst/lang/io/file/FileExists.kt
package org.dedda.bratwurst.lang.io.file

import org.dedda.bratwurst.lang.BWExpression
import org.dedda.bratwurst.lang.BWInteger
import org.dedda.bratwurst.lang.BWObject
import org.dedda.bratwurst.lang.BWString
import org.dedda.bratwurst.lang.ValueType.INTEGER
import org.dedda.bratwurst.lang.scope.Scope
import java.io.File

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
class FileExists(lineNumber: Int, private val variableName: String) : BWExpression(lineNumber) {
    private var exists = false
    override fun getValue(): BWObject {
        return BWInteger(intValue)
    }

    override fun getIntValue(): Int {
        return if (exists) 1 else 0
    }

    override fun getValueType() = INTEGER

    override fun run(scope: Scope) {
        val variable = scope.getVariable(variableName)
        if (variable.value !is BWString) {
            throw RuntimeException("variable not of type string!")
        }
        val fileName = (variable.value as BWString).stringValue
        val file = File(fileName)
        exists = file.exists()
    }

}